Bottom: c9f45a2741423bd36d9ad6d55a8041508e32ad97
Top:    b7723007a3de90e31e529427615b15cdffa567fa
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-23 12:06:51 -0400

Temporary holding patch


---

diff --git a/sound/soc/fsl/mpc5200_dma.c b/sound/soc/fsl/mpc5200_dma.c
index 95df860..a9ad54d 100644
--- a/sound/soc/fsl/mpc5200_dma.c
+++ b/sound/soc/fsl/mpc5200_dma.c
@@ -415,7 +415,7 @@ struct snd_soc_platform mpc5200_audio_dma_platform = {
 EXPORT_SYMBOL_GPL(mpc5200_audio_dma_platform);
 
 /* ---------------------------------------------------------------------
- * Sysfs attributes for debugging
+ * Sysfs attributes for error monitoring
  */
 
 static ssize_t psc_dma_status_show(struct device *dev,
@@ -590,7 +590,8 @@ int mpc5200_audio_dma_create(struct of_device *op)
 	if (rc)
 		dev_info(psc_dma->dev, "error creating sysfs files\n");
 
-	return 0;
+	/* Tell the ASoC OF helpers about it */
+	return snd_soc_register_platform(&mpc5200_audio_dma_platform);
 }
 EXPORT_SYMBOL_GPL(mpc5200_audio_dma_create);
 
@@ -600,6 +601,8 @@ int mpc5200_audio_dma_destroy(struct of_device *op)
 
 	dev_dbg(&op->dev, "mpc5200_audio_dma_destroy()\n");
 
+	snd_soc_unregister_platform(&mpc5200_audio_dma_platform);
+
 	bcom_gen_bd_rx_release(psc_dma->capture.bcom_task);
 	bcom_gen_bd_tx_release(psc_dma->playback.bcom_task);
 
@@ -609,7 +612,6 @@ int mpc5200_audio_dma_destroy(struct of_device *op)
 	free_irq(psc_dma->playback.irq, &psc_dma->playback);
 
 	iounmap(psc_dma->psc_regs);
-	iounmap(psc_dma->fifo_regs);
 	kfree(psc_dma);
 	dev_set_drvdata(&op->dev, NULL);
 
@@ -617,19 +619,6 @@ int mpc5200_audio_dma_destroy(struct of_device *op)
 }
 EXPORT_SYMBOL_GPL(mpc5200_audio_dma_destroy);
 
-static int __init mpc5200_soc_platform_init(void)
-{
-	/* Tell the ASoC OF helpers about it */
-	return snd_soc_register_platform(&mpc5200_audio_dma_platform);
-}
-module_init(mpc5200_soc_platform_init);
-
-static void __exit mpc5200_soc_platform_exit(void)
-{
-	snd_soc_unregister_platform(&mpc5200_audio_dma_platform);
-}
-module_exit(mpc5200_soc_platform_exit);
-
 MODULE_AUTHOR("Grant Likely <grant.likely@secretlab.ca>");
 MODULE_DESCRIPTION("Freescale MPC5200 PSC in DMA mode ASoC Driver");
 MODULE_LICENSE("GPL");
diff --git a/sound/soc/fsl/mpc5200_psc_i2s.c b/sound/soc/fsl/mpc5200_psc_i2s.c
index fd70143..f3c1fad 100644
--- a/sound/soc/fsl/mpc5200_psc_i2s.c
+++ b/sound/soc/fsl/mpc5200_psc_i2s.c
@@ -3,6 +3,7 @@
  * ALSA SoC Digital Audio Interface (DAI) driver
  *
  * Copyright (C) 2008 Secret Lab Technologies Ltd.
+ * Copyright (C) 2009 Jon Smirl, Digispeaker
  */
 
 #include <linux/module.h>
@@ -13,8 +14,6 @@
 #include <sound/pcm_params.h>
 #include <sound/soc.h>
 
-#include <sysdev/bestcomm/bestcomm.h>
-#include <sysdev/bestcomm/gen_bd.h>
 #include <asm/time.h>
 #include <asm/mpc52xx.h>
 #include <asm/mpc52xx_psc.h>
diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index bad9d88..cc1e618 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -20,6 +20,8 @@
  *   o Support TDM on PCM and I2S
  */
 
+#define DEBUG
+
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/init.h>
@@ -125,7 +127,7 @@ static int soc_pcm_apply_symmetry(struct snd_pcm_substream *substream)
 
 	if (codec_dai->symmetric_rates || cpu_dai->symmetric_rates ||
 	    machine->symmetric_rates) {
-		dev_dbg(card->dev, "Symmetry forces %dHz rate\n", 
+		dev_dbg(card->dev, "Symmetry forces %dHz rate\n",
 			machine->rate);
 
 		ret = snd_pcm_hw_constraint_minmax(substream->runtime,
diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index c1222fa..f3ccbd0 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -29,6 +29,8 @@
  *    o Support for reduced codec bias currents.
  */
 
+#define DEBUG
+
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/init.h>
